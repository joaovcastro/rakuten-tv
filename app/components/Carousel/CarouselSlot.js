import styled from 'styled-components';
import PropTypes from 'prop-types';

const CarouselSlot = styled.div`
  flex: 1 0 100%;
  flex-basis: 15%;
  margin-right: 20px;
  order: ${props => props.order};
`;

CarouselSlot.propTypes = {
  order: PropTypes.number.isRequired,
};

export default CarouselSlot;
