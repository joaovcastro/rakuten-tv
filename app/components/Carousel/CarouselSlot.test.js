import React from 'react';
import { shallow } from 'enzyme';

import CarouselSlot from './CarouselSlot';

describe('<CarouselSlot />', () => {
  it('correctly renders', () => {
    const wrapper = shallow(<CarouselSlot order={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
