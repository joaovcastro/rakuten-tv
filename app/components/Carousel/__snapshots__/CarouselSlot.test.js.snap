// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselSlot /> correctly renders 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  flex: 1 0 100%;
  flex-basis: 15%;
  margin-right: 20px;
  order: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "order": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  order={1}
/>
`;
